flashlight/fl/examples/RnnLm.cpp

```diff
@@ -79,41 +79,66 @@ class LMDataset : public Dataset {
 class RnnLm : public Container {
  public:
   explicit RnnLm(int vocab_size, int hidden_size = 200)
-      : embed(Embedding(hidden_size, vocab_size)),
-        rnn(
-            RNN(hidden_size,
-                hidden_size,
-                2, /* Num layers. */
-                RnnMode::LSTM,
-                false /* Dropout */)),
-        linear(Linear(hidden_size, vocab_size)) {
+      : embed(std::make_shared<Embedding>(hidden_size, vocab_size)),
+        rnn(std::make_shared<RNN>(
+            hidden_size,
+            hidden_size,
+            2, /* Num layers. */
+            RnnMode::LSTM,
+            0 /* Dropout */)),
+        linear(std::make_shared<Linear>(hidden_size, vocab_size)),
+        logsoftmax_(0) // max on the main dimension
+  {
+    createLayers();
+  }
+
+  RnnLm(const RnnLm& other) {
+    copy(other);
+    createLayers();
+  }
+
+  RnnLm(RnnLm&& other) = default;
```

Comment:
the default the compiler uses should be fine, no? This line shouldn't be needed?